diffraction/remove_prompt_pulses.py

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+
+import numpy as np
+import scipp as sc
+import scippneutron as scn
+
+
+def calculate_pulse_times(tmin, tmax, period):
+    """
+    Calculates the intervals of tof to be used for removing pulses in remove_prompt_pulses
+
+    tmin: minimum time-of-flight in microsecconds
+
+    tmax: maximum time-of-flight in microseconds
+
+    period: period of the source in microseconds
+    """
+    times = []
+    time = 0.
+    # find when the first prompt pulse would be
+    while time < tmin:
+        time += period
+    # calculate all times possible
+    while time < tmax:
+        times.append(time)
+        time += period
+    return times
+
+
+def remove_prompt_pulses(dataset, width, frequency=None):
+    """
+    Removes the prompt pulse for a time of flight measurement.
+
+    dataset: input scipp Dataset
+
+    width: width of the time-of-flight (in microseconds) to remove from the data
+
+    frequency: frequency of the source (in Hz) used to calculate the minimum time 
+               of flight to filter.
+               If not provided it is calculated from the attribute in the dataset
+    """
+    # check inputs
+    assert width > 0, 'the width has to be positive'
+
+    # dataset must have tof as 1 of its coordinates
+    assert 'tof' in list(dataset.dims), 'tof must be one of the coordinates of dataset'
+    assert dataset.coords['tof'].unit == sc.units.us, 'tof must be expressed in microseconds'
```

Comment:
Do not require a specific unit, instead use `tof = sc.to_unit(da.coords['tof'], sc.units.us, copy=False)`.
Or, even better if possible, do not specify any units and let scipp figure everything out. This only works if you can avoid using numpy or raw numbers.